Libraries/LibGC/ForeignCell.h

```diff
@@ -0,0 +1,166 @@
+/*
+ * Copyright (c) 2024, Andrew Kaster <andrew@ladybird.org>
+ *
+ * SPDX-License-Identifier: BSD-2-Clause
+ */
+
+#pragma once
+
+#include <AK/String.h>
+#include <AK/Swift.h>
+#include <AK/TypeCasts.h>
+#include <LibGC/Cell.h>
+#include <LibGC/DeferGC.h>
+
+namespace GC {
+
+#define FOREIGN_CELL(class_, base_class) \
+    using Base = base_class;             \
+    friend class GC::Heap;
+
+class ForeignCell : public Cell {
+    FOREIGN_CELL(ForeignCell, Cell);
+
+public:
+    struct ForeignClass {
+        String class_name; // FIXME: FlyString?
```

Comment:
I know you're in the get-it-working phase and not the make-it-faster phase, but at least for Swift it's pretty straightforward to get the type name from type metadata (cast pointer to `Any.Type`, the call `String(describing:)`) on demand, so "get the class name" could be an operation in the vtable.

Reply:
Right, I don't think we can get around having to have our own storage of the class name. Eventually we should cache the vtable, possibly keyed on the name, in which case it would make more sense for us to keep it around.

Reply:
The type metadata pointer (i.e., the `Any.Type`) is globally uniqued by the Swift runtime and is immortal, so it can be used as a key to uniquely identify the type within the process. If it's just a unique representation of the Swift type you need, I recommend using that rather than a string. Up to you, of course!